### PR TITLE
Add configuration for file paths (fixes #722)

### LINKED ISSUE #722: Make dirs configurable via pikaur.conf
yeah it could be useful to make those dir paths (https://github.com/actionless/pikaur#file-locations) configurable via pikaur.conf

_Originally posted by @actionless in https://github.com/actionless/pikaur/discussions/715#discussioncomment-6006767_

### PR DESCRIPTION
**Pull request by samuel-jimenez**
Fixes #722

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,6 +39,7 @@ Pikaur wraps all the pacman options accurately except for `-Syu` which is being
 * [Installation](#installation "")
 * [Run without installation](#run-without-installation "")
 * [File locations](#file-locations "")
+* [Options](#options "")
```
Comment (actionless) [inline README.md:42]: please use the same help system, as all the other existing cli options

File: README.md [hunk 2]
```diff
@@ -223,6 +239,13 @@ User ID to run makepkg if pikaur started from root.
 Setting this option would override DynamicUsers settings and force changing to this UID instead of a dynamic one.
 
 
+##### CachePath (default: ~/.cache)
+Path to package cache location.
```
Comment (actionless) [inline README.md:243]: mention that those config options would be overridden by those cli flags

**Comment by samuel-jimenez** [comment]
Documentation updated

File: README.md [hunk 3]
```diff
@@ -31,11 +31,18 @@ The following pacman operations are extended with AUR capabilities:
 * `-Sc` / `-Scc` (build dir/built packages cache clean)
 * `-Qu` (query upgradeable, `-q` supported)
 
+Pikaur also provides the following additional command line arguments:
```
Comment (actionless) [inline README.md:34]: no, it's still not the way how all the options are documented, those couple of flags above are just usage examples, full documentation for flags should be only in the code (ie when --help is passed) and shouldn't be repeated in the readme

**Comment by actionless** [comment]
also you need to default cli flags to the value from the config,
see implementation for --preserve-env as an example: https://github.com/actionless/pikaur/blob/master/pikaur/args.py#L256-L260

File: pikaur/config.py [hunk 4]
```diff
@@ -151,7 +151,7 @@ def init_value(cls) -> Path:
             or os.environ.get(
                 "XDG_CACHE_HOME",
             )
-            or Home()() / ".cache/",
+            or Path(PikaurConfig().misc.CachePath.get_str()),
```
Comment (actionless) [inline pikaur/config.py:154]: the order is wrong - if user have XDG_CACHE_HOME set - the config value would be ignored.

Comment (actionless) [reply pikaur/config.py:154]: or otherwise, the documentation should be updated to mention that config option would be overriden by this specific env var

**Comment by samuel-jimenez** [comment]
Changes made.

File: README.md [hunk 5]
```diff
@@ -70,8 +69,6 @@ cd pikaur
 python3 ./pikaur.py -S AUR_PACKAGE_NAME
 ```
 
-
-
```
Comment (actionless) [inline README.md:74]: no need to change the formatting here

Comment (samuel-jimenez) [reply README.md]: fixed

**Comment by actionless** [comment]
thanks for your first contribution! 🎉

**Pull request merged**